File: app/src/main/java/com/perflyst/twire/activities/stream/VODActivity.java
package com.perflyst.twire.activities.stream;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

import com.perflyst.twire.R;
import com.perflyst.twire.activities.ChannelActivity;
import com.perflyst.twire.model.VideoOnDemand;

public class VODActivity extends StreamActivity {
    private VideoOnDemand mVod;
    private Fragment vodsFragments;
    private TextView mTitleView, mViewsView;

    public static Intent createVODIntent(VideoOnDemand video, Context context, boolean transition) {
        Intent intent = new Intent(context, VODActivity.class);
        intent.putExtra(context.getString(R.string.intent_vod), video);
        intent.putExtra(context.getString(R.string.stream_shared_transition), transition);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);

        return intent;
    }

    @Override
    protected int getLayoutResource() {
        return R.layout.activity_vod;
    }

    @Override
    protected int getVideoContainerResource() {
        return R.id.video_fragment_container;
    }

    @Override
    protected Bundle getStreamArguments() {
        if (mVod == null) {
            Intent intent = getIntent();
            mVod = intent.getParcelableExtra(getString(R.string.intent_vod));
        }

        Bundle args = new Bundle();
        args.putParcelable(getString(R.string.stream_fragment_streamerInfo), mVod.channelInfo);
        args.putString(getString(R.string.stream_fragment_vod_id), mVod.videoId);
        args.putString(getString(R.string.stream_fragment_title), mVod.videoTitle);
        return args;
    }

    @Override
    protected void onCreate(Bundle savedInstance) {
        super.onCreate(savedInstance);
        if (savedInstance == null) {
            FragmentManager fm = getSupportFragmentManager();

            if (vodsFragments == null) {
                vodsFragments = ChannelActivity.VodFragment.newInstance(mVod.isBroadcast, mVod.channelInfo);
                fm.beginTransaction().replace(R.id.additional_vods_container, vodsFragments).commit();
            }
        }

        mTitleView = findViewById(R.id.title);
        mViewsView = findViewById(R.id.views);

        setVodData();
    }

    private void setVodData() {
        if (mVod != null) {
            mTitleView.setText(mVod.videoTitle);
            mViewsView.setText(getString(R.string.vod_views, mVod.views));
        }
    }

    public void startNewVOD(VideoOnDemand videoOnDemand) {
        mVod = videoOnDemand;
        setVodData();
        resetStream();
    }
}

File: app/src/main/java/com/perflyst/twire/activities/stream/VODActivity.kt
package com.perflyst.twire.activities.stream

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.perflyst.twire.R
import com.perflyst.twire.activities.ChannelActivity.VodFragment.Companion.newInstance
import com.perflyst.twire.model.VideoOnDemand

class VODActivity : StreamActivity() {
    private var mVod: VideoOnDemand? = null
    private var vodsFragments: Fragment? = null
    private var mTitleView: TextView? = null
    private var mViewsView: TextView? = null

    override val layoutResource: Int get() = R.layout.activity_vod

    override val videoContainerResource: Int get() = R.id.video_fragment_container

    override val streamArguments: Bundle
        get() {
            if (mVod == null) {
                val intent = getIntent()
                mVod = intent.getParcelableExtra(getString(R.string.intent_vod))
            }

            val args = Bundle()
            args.putParcelable(getString(R.string.stream_fragment_streamerInfo), mVod!!.channelInfo)
            args.putString(getString(R.string.stream_fragment_vod_id), mVod!!.videoId)
            args.putString(getString(R.string.stream_fragment_title), mVod!!.videoTitle)
            return args
        }

    override fun onCreate(savedInstance: Bundle?) {
        super.onCreate(savedInstance)
        if (savedInstance == null) {
            val fm = supportFragmentManager

            if (vodsFragments == null) {
                vodsFragments = newInstance(mVod!!.isBroadcast, mVod!!.channelInfo)
                fm.beginTransaction().replace(R.id.additional_vods_container, vodsFragments!!)
                    .commit()
            }
        }

        mTitleView = findViewById(R.id.title)
        mViewsView = findViewById(R.id.views)

        setVodData()
    }

    private fun setVodData() {
        if (mVod != null) {
            mTitleView!!.text = mVod!!.videoTitle
            mViewsView!!.text = getString(R.string.vod_views, mVod!!.views)
        }
    }

    fun startNewVOD(videoOnDemand: VideoOnDemand?) {
        mVod = videoOnDemand
        setVodData()
        resetStream()
    }

    companion object {
        @JvmStatic
        fun createVODIntent(video: VideoOnDemand?, context: Context, transition: Boolean): Intent {
            val intent = Intent(context, VODActivity::class.java)
            intent.putExtra(context.getString(R.string.intent_vod), video)
            intent.putExtra(context.getString(R.string.stream_shared_transition), transition)
            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)

            return intent
        }
    }
}
